### PR TITLE
Add initial implementation of mutating admission controller webhook

### PR DESCRIPTION
**Pull request by bpradipt**
Fixes: #73 
Signed-off-by: Pradipta Banerjee <pradipta.banerjee@gmail.com>

### CODE REVIEW

File: webhook/pkg/mutating_webhook/remove-resourcespec.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+const (
+	DEFAULT_RUNTIME_CLASS_NAME  = "kata-remote-cc"
+	VM_ANNOTATION_INSTANCE_TYPE = "kata.peerpods.io/instance_type"
+	VM_INSTANCE_TYPE_DEFAULT    = "t2.small"
+	VM_EXTENDED_RESOURCE        = "kata.peerpods.io/vm"
```
Comment (stevenhorsman) [inline webhook/pkg/mutating_webhook/remove-resourcespec.go:14]: Not a big problem for now, but I wonder if we should move away from kata to cc/confidential-containers?

**Review by stevenhorsman — APPROVED** [review]
Hey @bpradipt, I've run through the documentation and I think I have it working as expected. I've left some minor comments/suggestions.
In addition, I'm slightly concerned that the `fixes #7` comment in the commit will close the issue and I think there is probably more to do, so we'll have to remember to check & re-open it.
I'll also admit that I'm far from being a Kubernetes expert, so @yoheiueda, or his team might be able to provide more value in review.
Thanks!

File: webhook/README.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+```
+2. Create the pod
+```sh
+kubectl apply -f hack/pod.yaml
+```
+3. View the mutated pod
+```sh
+kubectl get -f hack/pod.yaml -o yaml | grep kata.peerpods
+```
+
```
[the remark this answers is not in the document]
Comment (bpradipt) [reply webhook/README.md:86]: Thanks @stevenhorsman will update the doc

File: webhook/pkg/mutating_webhook/remove-resourcespec.go (hunk 1 above)
Comment (bpradipt) [reply webhook/pkg/mutating_webhook/remove-resourcespec.go:14]: Yeah, this is a good suggestion. I'll make the changes

File: webhook/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+## Getting Started
+You’ll need a Kubernetes cluster to run against. You can use [KIND](https://sigs.k8s.io/kind) to get a local cluster for testing, or run against a remote cluster.
+
+You'll also need to advertise the extended resource `kata.peerpods.io/vm`.
+
+A simple daemonset is provided under the following [directory](./hack/extended-resources/).
+
+```
+./hack/extended-resources/setup.sh
```
[the remark this answers is not in the document]
Comment (bpradipt) [reply webhook/README.md:33]: @stevenhorsman good catch on the #7. I think its better to keep #7 open. I'll create a new issue specifically mentioning mutation admission controller implementation and link this PR.

File: webhook/pkg/mutating_webhook/remove-resourcespec.go (hunk 1 above)
Comment (bpradipt) [reply webhook/pkg/mutating_webhook/remove-resourcespec.go:14]: @stevenhorsman re-visiting the extended resource naming aspect, I'm thinking of leaving it as-is for now and get some more ideas before finalising the name :-)

**Review by yoheiueda — APPROVED** [review]
LGTM

**Pull request merged**